include/graaflib/algorithm/coloring/greedy_graph_coloring.tpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <graaflib/algorithm/coloring/greedy_graph_coloring.h>
+
+#include <unordered_map>
+
+namespace graaf::algorithm {
+
+template <typename GRAPH>
+std::unordered_map<vertex_id_t, int> greedy_graph_coloring(const GRAPH& graph) {
+  // Initialize a map to store the coloring
+  std::unordered_map<vertex_id_t, int> coloring;
+
+  // Get the vertices from the graph
+  const auto& vertices = graph.get_vertices();
+
+  // Iterate through each vertex
+  for (const auto& vertex_pair : vertices) {
+    const vertex_id_t& current_vertex_id = vertex_pair.first;
```

Comment:
[minor]: Another way of doing this would be to use structured binding to unpack the pair:

```suggestion
  for (const auto& [current_vertex_id, _] : vertices) {
```